### PR TITLE
[DOCS] Adds KNN object sub-properties individually to common params

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR:
* adds the sub-properties of the kNN object individually to the common parameters so that they can be referenced separately,
* removes `query_vector` kNN sub-property from the semantic search API docs.

### Preview

* [Semantic search](https://elasticsearch_91503.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-api.html)
* [kNN search](https://elasticsearch_91503.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/knn-search-api.html)
* [Search](https://elasticsearch_91503.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/search-search.html)

### CODE REVIEW

**Review by benwtrent — APPROVED** [review]
Looks good from Search side.

File: docs/reference/search/knn-search.asciidoc [hunk 1]
```diff
@@ -100,6 +100,30 @@ include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=routing]
...
 include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=knn]
++
+.Properties of `knn` object
+[%collapsible%open]
+====
+`field`::
+(Required, string) 
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=knn-field]
+
+`filter`::
```
Comment (davidkyle) [inline docs/reference/search/knn-search.asciidoc:111]: For the old and now deprecated _knn_search API the `filter` is at the same level as the `knn` object and should be lifted out of here. 

https://www.elastic.co/guide/en/elasticsearch/reference/8.4/knn-search-api.html#knn-search-api-request-body

For knn as part of _search and _semantic_search `filter` is part of `knn` which is correct. 

https://www.elastic.co/guide/en/elasticsearch/reference/8.4/search-search.html#search-api-knn

So we just need to change it here for _knn_search

Comment (szabosteve) [reply docs/reference/search/knn-search.asciidoc]: Fixed in [1d63e86](https://github.com/elastic/elasticsearch/pull/91503/commits/1d63e86c31160f6a5355f23a19301e07a42c44c8).

**Review by davidkyle — APPROVED** [review]
LGTM

Thanks for changing this I didn't realise all 3 were slightly different

**Pull request merged**